### PR TITLE
Added bag part indexing and fixed simple query search.

### PR DESCRIPTION
**Pull request by hswami**
Fixed issue [#2431: Content items in a bag part are not indexed.](https://github.com/OrchardCMS/OrchardCore/issues/2431)
Fixed issue [#2437: Search type "simple_query_string" throws an exception.](https://github.com/OrchardCMS/OrchardCore/issues/2437)

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Indexing.Abstractions/ContentPartIndexHandler.cs [hunk 1]
```diff
@@ -10,18 +10,18 @@ namespace OrchardCore.Indexing
     /// </summary>
     public abstract class ContentPartIndexHandler<TPart> : IContentPartIndexHandler where TPart : ContentPart
     {
-        Task IContentPartIndexHandler.BuildIndexAsync(ContentPart contentPart, ContentTypePartDefinition typePartDefinition, BuildIndexContext context, ContentIndexSettings settings)
+        public async Task BuildIndexAsync(ContentPart contentPart, ContentTypePartDefinition typePartDefinition, BuildIndexContext context, ContentIndexSettings settings)
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Indexing.Abstractions/ContentPartIndexHandler.cs]: Why are you adding `async`. It's better not to use the keyword when not necessary. Like in this case where the only statement to await is returned.

File: src/OrchardCore.Modules/OrchardCore.Flows/Indexing/BagPartIndexHandler.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                // Lazy resolution to prevent cyclic dependency
+                var contentItemIndexHandlers = _serviceProvider.GetServices<IContentItemIndexHandler>();
+
+                foreach (var contentItemIndexHandler in contentItemIndexHandlers)
+                {
+                    foreach (var contentItem in bagPart.ContentItems)
+                    {
+                        var buildIndexContext = new BuildIndexContext(context.DocumentIndex, contentItem, contentItem.ContentType);
+
+                        await contentItemIndexHandler.BuildIndexAsync(buildIndexContext);
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Flows/Indexing/BagPartIndexHandler.cs]: When a bag part content item is indexed, shouldn't the keys be prefixed by the container keys?

**Comment by hswami** [comment]
Some of the files are not UTF-8 encoded with appropriate BOMs as they should be. Closing this PR to fix. Will resubmit.

File: src/OrchardCore/OrchardCore.Lucene.Core/QueryProviders/SimpleQueryStringQueryProvider.cs [hunk 3]
```diff
@@ -14,9 +15,19 @@ public Query CreateQuery(ILuceneQueryService builder, LuceneQueryContext context
             }
 
             var queryString = query["query"]?.Value<string>();
-            var fields = query["fields"]?.Value<string>() ?? "";
-            var defaultOperator = query["default_operator"]?.Value<string>() ?? "and";
-            var queryParser = new SimpleQueryParser(context.DefaultAnalyzer, "");
+            var fields = query["fields"]?.Values<string>() ?? new string[0];
+            var defaultOperator = query["default_operator"]?.Value<string>().ToLowerInvariant() ?? "or";
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Lucene.Core/QueryProviders/SimpleQueryStringQueryProvider.cs]: I would prefer not to change the default. I assume that the more terms you add, the less results you get. Unless the Elastic search states the opposite.

File: src/OrchardCore/OrchardCore.Indexing.Abstractions/ContentPartIndexHandler.cs (hunk 1 above)
Comment (hswami) [reply src/OrchardCore/OrchardCore.Indexing.Abstractions/ContentPartIndexHandler.cs]: I'll remove the async.

**Pull request closed without merging**